src/xinterpreter.cpp

```diff
@@ -131,6 +131,42 @@ namespace xeus
         }
     }
 
+    nl::json xinterpreter::publish_execution_output(int execution_count,
+                                                const std::string& status,
+                                                const std::string& ename,
+                                                const std::string& evalue,
+                                                const std::string& data_type,
+                                                const std::vector<std::string>& data_content,
+                                                const std::vector<std::string>& trace_back)
```

Comment:
I think it makes sense to have default arguments for some of these variables. e.g trace_back, all msgs that don't have the `error` `status` type should receive an empty array.